syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.Experimental;
option java_package = "com.yandex.ydb.experimental";
option java_outer_classname = "ExperimentalProtos";

import "ydb/public/api/protos/ydb_issue_message.proto"; 
import "ydb/public/api/protos/ydb_status_codes.proto"; 
import "ydb/public/api/protos/ydb_value.proto"; 


message ExecuteStreamQueryRequest {
    string yql_text = 1;
    map<string, TypedValue> parameters = 2;

    enum ProfileMode {
        PROFILE_MODE_UNSPECIFIED = 0;
        NONE = 1;
        BASIC = 2;
        FULL = 3;
    }

    ProfileMode profile_mode = 3;
    bool explain = 4;
}

message ExecuteStreamQueryResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
    ExecuteStreamQueryResult result = 3;
}

message StreamQueryProgress {
}

message ExecuteStreamQueryResult {
    oneof result {
        Ydb.ResultSet result_set = 1;
        string profile = 2;
        StreamQueryProgress progress = 3;
        string query_plan = 4;
    }
}
